modules/telemetry/telemetrymodule.cpp

```diff
@@ -0,0 +1,314 @@
+    addProperty(_port);
+
+    // Add options to the drop down menu
+    _modeOptions.addOptions({
+        { 0, "Horizontal" },
+        { 1, "Circular" }
+    });
+    _modeOptions.onChange([this]() { guiOnChangeAngleCalculationMode(); });
+
+    // Select Horizontal angle calculation mode as the default
+    _modeOptions.setValue(static_cast<int>(AngleCalculationMode::Horizontal));
+    addProperty(_modeOptions);
+
+    addProperty(_includeElevationAngle);
+}
+
+TelemetryModule::~TelemetryModule() {
+    // Clear the telemetries list
+    for (TelemetryBase* telemetry : _telemetries) {
+        delete telemetry;
+    }
+}
+
+void TelemetryModule::guiOnChangeAngleCalculationMode() {
+    _angleCalculationMode = static_cast<AngleCalculationMode>(_modeOptions.value());
+}
+
+void TelemetryModule::internalInitialize(const ghoul::Dictionary& dictionary) {
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _ipAddress = p.ipAddress.value_or(_ipAddress);
+    _port = p.port.value_or(_port);
+
+    if (p.angleCalculationMode.has_value()) {
+        Parameters::AngleCalculationMode mode =
+            Parameters::AngleCalculationMode(*p.angleCalculationMode);
+        _angleCalculationMode = codegen::map<AngleCalculationMode>(mode);
+    }
+
+    // Fill telemetry list
+    TelemetryBase* telemetry = new AngleModeTelemetry(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    telemetry = new CameraTelemetry(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    telemetry = new FocusTelemetry(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    telemetry = new TimeTelemetry(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    telemetry = new NodesTelemetry(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    telemetry = new PlanetsCompareSonification(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    telemetry = new PlanetsOverviewSonification(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    telemetry = new PlanetsSonification(_ipAddress, _port);
+    addTelemetry(telemetry);
+
+    // Only the master runs the TelemetryModule update thread
+    if (global::windowDelegate->isMaster()) {
+        _isRunning = true;
+        _updateThread = std::thread([this]() { update(std::ref(_isRunning)); });
+
+        // Make sure the telemetry thread is synced with the main thread
+        global::callback::postSyncPreDraw->emplace_back([this]() {
+            // Tell the telemetry thread that a new frame is starting
+            //LDEBUG("The main thread signals to the telemetry thread");
+            syncToMain.notify_one();
+        });
+    }
+}
+
+void TelemetryModule::addTelemetry(TelemetryBase* telemetry) {
+    _telemetries.push_back(telemetry);
+    addPropertySubOwner(telemetry);
+}
+
+void TelemetryModule::internalDeinitialize() {
+    // Stop the loop before joining and tell the thread it is ok to run last itteration
+    _isRunning = false;
+    syncToMain.notify_one();
+
+    // Join the thread
```

Comment:
This comment seems superfluous, unless it is elaborated on, e.g. about what "the thread" is